pkg/discovery/dtofactory/group_dto_builder.go

```diff
@@ -63,77 +70,146 @@ func (builder *groupDTOBuilder) BuildGroupDTOs() []*proto.GroupDTO {
+// These group members have resize consistent flag set to true.
+func (builder *groupDTOBuilder) createSubGroups(entityGroup *repository.EntityGroup,
+	entityType metrics.DiscoveredEntityType) []*proto.GroupDTO {
 
-			// static group
-			groupBuilder := group.StaticGroup(id).
-				OfType(protoType).
-				WithEntities(containerList).
-				WithDisplayName(displayName)
+	var result []*proto.GroupDTO
+	if entityType != metrics.ContainerType {
+		return []*proto.GroupDTO{}
+	}
 
-			// if parent group resize policy is consistent resize, set it for the container sub-group
-			_, exists := CONSISTENT_RESIZE_GROUPS[entityGroup.ParentKind]
-			if exists && entityGroup.ParentName != "" {
-				glog.V(2).Infof("%s: set group to resize consistently\n", groupId)
-				groupBuilder.ResizeConsistently()
-			}
+	// Only one type of container in the pod, so sub groups are not created.
+	// In this case, the parent level container group has the consistent resize flag set to true
+	if len(entityGroup.ContainerGroups) <= 1 {
+		return []*proto.GroupDTO{}
+	}
 
-			// build group
-			groupDTO, err := groupBuilder.Build()
-			if err != nil {
-				glog.Errorf("Error creating group dto  %s::%s", id, err)
-				continue
-			}
+	// Additional sub groups for the different containers running in a pod
+	for containerName, containerList := range entityGroup.ContainerGroups {
+		etype := metrics.ContainerType
+		groupId := fmt.Sprintf("%s::%s", entityGroup.GroupId, containerName)
+		protoType := proto.EntityDTO_CONTAINER
 
-			result = append(result, groupDTO)
+		// resize policy setting based on the parent type of the group
+		resizeFlag := consistentResize(entityGroup)
+		subGroup := builder.createGroup(entityGroup, etype, groupId, protoType, containerList, resizeFlag)
 
-			glog.V(4).Infof("groupDTO  : %++v", groupDTO)
+		if subGroup != nil {
+			result = append(result, subGroup)
 		}
 	}
 
 	return result
 }
+
+// Create a static group for pod or container
+func (builder *groupDTOBuilder) createGroup(entityGroup *repository.EntityGroup, entityType metrics.DiscoveredEntityType,
+	groupId string, protoType proto.EntityDTO_EntityType,
+	memberList []string, resizeFlag bool) *proto.GroupDTO {
+
+	// group id created using the parent type, name and target identifier
+	id := fmt.Sprintf("%s-%s[%s]", groupId, builder.targetId, entityType)
+	displayName := fmt.Sprintf("%ss By %s [%s]", entityType, groupId, builder.targetId)
+
+	// static group
+	groupBuilder := group.StaticGroup(id).
+		OfType(protoType).
+		WithEntities(memberList).
+		WithDisplayName(displayName)
+
+	// resize flag setting
+	if resizeFlag {
+		glog.V(4).Infof("%s: set group to resize consistently", displayName)
+		groupBuilder.ResizeConsistently()
+	}
+
+	// build group
+	groupDTO, err := groupBuilder.Build()
+	if err != nil {
+		glog.Errorf("Error creating group dto  %s::%s", id, err)
+		return nil
+	}
+
+	return groupDTO
+}
+
+func consistentResize(entityGroup *repository.EntityGroup) bool {
```

Comment:
Can we simplify this function? We don't need to check empty string, for example, we can do
```
func consistentResize(entityGroup *repository.EntityGroup) bool {
        _, resize := CONSISTENT_RESIZE_GROUPS[entityGroup.ParentKind]
        return resize
}
```